src/overlays/actors/ovl_En_Rat/z_en_rat.c

```diff
@@ -28,123 +47,913 @@ const ActorInit En_Rat_InitVars = {
+            Actor_MoveWithGravity(&this->actor);
+            Actor_UpdateBgCheckInfo(play, &this->actor, 20.0f, 30.0f, 60.0f, 7);
+        }
+
+        if (SurfaceType_IsWallDamage(&play->colCtx, this->actor.floorPoly, this->actor.floorBgId)) {
+            EnRat_Explode(this, play);
+            return;
+        }
+
+        this->collider.dim.worldSphere.center.x = this->actor.world.pos.x + (this->axisUp.x * 10.0f);
+        this->collider.dim.worldSphere.center.y = this->actor.world.pos.y + (this->axisUp.y * 10.0f);
+        this->collider.dim.worldSphere.center.z = this->actor.world.pos.z + (this->axisUp.z * 10.0f);
+
+        if (this->actionFunc != EnRat_Revive) {
+            if (this->damageReaction.stunTimer == 0) {
+                CollisionCheck_SetAT(play, &play->colChkCtx, &this->collider.base);
+            }
+
+            CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+            CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+        }
+
+        Actor_SetFocus(&this->actor, this->actor.shape.yOffset * 0.015f);
+    }
+}
+
+s32 EnRat_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnRat* this = THIS;
+
+    if (limbIndex == REAL_BOMBCHU_LIMB_BODY) {
+        pos->y -= this->revivePosY;
+    }
+
+    if (limbIndex == REAL_BOMBCHU_LIMB_TAIL_END) {
+        *dList = NULL;
+    }
+
+    return false;
+}
+
+void EnRat_PostLimbDraw(PlayState* play2, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    PlayState* play = play2;
+    EnRat* this = THIS;
+    MtxF* currentMatrixState;
+    Vec3f* ptr;
+    f32 redModifier;
+
+    if (limbIndex == REAL_BOMBCHU_LIMB_TAIL_END) {
+        OPEN_DISPS(play->state.gfxCtx);
+
+        Matrix_ReplaceRotation(&play->billboardMtxF);
+        Matrix_MultZero(&this->smokePos);
+        this->smokePos.y += 15.0f;
+        currentMatrixState = Matrix_GetCurrent();
+
+        if (this->actionFunc == EnRat_ChasePlayer) {
+            s32 i;
+
+            gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, 150, 255);
+            gDPSetEnvColor(POLY_XLU_DISP++, 0xFF, 0x00, 0x00, 0x00);
+            Matrix_Scale(45.0f, 45.0f, 45.0f, MTXMODE_APPLY);
+
+            for (i = 0; i < ARRAY_COUNT(this->sparkOffsets); i++) {
+                ptr = &this->sparkOffsets[i];
+                currentMatrixState->mf[3][0] = this->smokePos.x + ptr->x;
+                currentMatrixState->mf[3][1] = this->smokePos.y + ptr->y;
+                currentMatrixState->mf[3][2] = this->smokePos.z + ptr->z;
+                gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx),
+                          G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                gSPSegment(POLY_XLU_DISP++, 0x08, sSparkTextures[(play->gameplayFrames + i) & 3]);
+                gSPDisplayList(POLY_XLU_DISP++, gEffSparkDL);
+            }
+
+            Matrix_Scale(0.022222223f, 0.022222223f, 0.022222223f, MTXMODE_APPLY);
```

Comment:
Could these be 1 / 45?

Reply:
Nope, that doesn't work.

Reply:
Hmm, this surprises me. My script says `1.0f/45.0f` should produce the same hex as in the rodata.

Reply:
Ah, apparently it works, but only if you replace all three. I replaced one, and it didn't work.